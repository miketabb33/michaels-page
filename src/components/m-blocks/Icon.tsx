import React from 'react'

export type IconName = 'rotate' | 'github' | 'linkedin'

type IconProps = {
  iconName: IconName
  className?: string
}

const iconMap = new Map<IconName, string>([
  ['rotate', '/images/icons/rotate.png'],
  ['github', '/images/social-media-icons/github.png'],
  ['linkedin', '/images/social-media-icons/linkedin.png'],
])

const Icon = ({ iconName, className }: IconProps) => {
  return <img src={iconMap.get(iconName)} alt={`${iconName}-icon`} className={className}></img>
}

export default Icon
