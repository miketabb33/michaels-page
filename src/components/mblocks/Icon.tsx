import React from 'react'

export type IconName = 'rotate'

type IconProps = {
  iconName: IconName
}

const iconMap = new Map<IconName, string>([
  ['rotate', 'images/icons/rotate.png'],
])

const Icon = ({ iconName }: IconProps) => {
  return <img src={iconMap.get(iconName)} alt={`${iconName}-icon`}></img>
}

export default Icon
