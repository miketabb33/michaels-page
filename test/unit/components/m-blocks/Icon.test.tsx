import { render } from '@testing-library/react'
import Icon from '../../../../src/components/m-blocks/Icon'
import React from 'react'

describe('Icon', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Icon iconName="rotate" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
