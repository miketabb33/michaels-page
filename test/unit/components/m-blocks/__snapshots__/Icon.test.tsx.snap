// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match snapshot 1`] = `
<DocumentFragment>
  <img
    alt="rotate-icon"
    src="images/icons/rotate.png"
  />
</DocumentFragment>
`;
